specs/phase1/beacon-chain.md

```diff
@@ -739,311 +433,117 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
 
-##### New default validator for deposits
+    # Verify signature
+    signer_index = get_shard_proposer_index(state, header.slot, header.shard)
+    signing_root = compute_signing_root(header, get_domain(state, DOMAIN_SHARD_HEADER))
+    assert bls.Verify(state.validators[signer_index].pubkey, signing_root, signed_header.signature)
 
-```python
-def get_validator_from_deposit(state: BeaconState, deposit: Deposit) -> Validator:
-    amount = deposit.data.amount
-    effective_balance = min(amount - amount % EFFECTIVE_BALANCE_INCREMENT, MAX_EFFECTIVE_BALANCE)
-    next_custody_secret_to_reveal = get_custody_period_for_validator(
-        ValidatorIndex(len(state.validators)),
-        get_current_epoch(state),
+    # Verify the length by verifying the degree.
+    if header.commitment.length == 0:
+        assert header.degree_proof == G1_SETUP[0]
+    assert (
+        bls.Pairing(header.degree_proof, G2_SETUP[0])
+        == bls.Pairing(header.commitment.point, G2_SETUP[-header.commitment.length]))
     )
 
-    return Validator(
-        pubkey=deposit.data.pubkey,
-        withdrawal_credentials=deposit.data.withdrawal_credentials,
-        activation_eligibility_epoch=FAR_FUTURE_EPOCH,
-        activation_epoch=FAR_FUTURE_EPOCH,
-        exit_epoch=FAR_FUTURE_EPOCH,
-        withdrawable_epoch=FAR_FUTURE_EPOCH,
-        effective_balance=effective_balance,
-        next_custody_secret_to_reveal=next_custody_secret_to_reveal,
-        all_custody_secrets_revealed_epoch=FAR_FUTURE_EPOCH,
-    )
+    # Get the correct pending header list
+    if compute_epoch_at_slot(header.slot) == get_current_epoch(state):
+        pending_headers = state.current_epoch_pending_shard_headers
+    else:
+        pending_headers = state.previous_epoch_pending_shard_headers
+
+    # Check that this header is not yet in the pending list
+    assert header_root not in [pending_header.root for pending_header in pending_headers]
+
+    # Include it in the pending list
+    index = compute_committee_index_from_shard(state, header.slot, header.shard)
+    committee_length = len(get_beacon_committee(state, header.slot, index))
+    pending_headers.append(PendingShardHeader(
+        slot=header.slot,
+        shard=header.shard,
+        commitment=header.commitment,
+        root=header_root,
+        votes=Bitlist[MAX_VALIDATORS_PER_COMMITTEE]([0] * committee_length),
+        confirmed=False,
+    ))
 ```
 
-#### Light client processing
+The degree proof works as follows. For a block `B` with length `l` (so `l`  values in `[0...l - 1]`, seen as a polynomial `B(X)` which takes these values), the length proof is the commitment to the polynomial `B(X) * X**(MAX_DEGREE + 1 - l)`, where `MAX_DEGREE` the the maximum power of `s` available in the setup, which is `MAX_DEGREE = len(G2_SETUP) - 1`. The goal is to ensure that a proof can only be constructed if `deg(B) < l` (there are not hidden higher-order terms in the polynomial, which would thwart reconstruction).
```

Comment:
possible typo:
"where `MAX_DEGREE` the the maximum power of `s`" -> "where `MAX_DEGREE` is the maximum power of `s`"

Reply:
yep, thank you!